labman/gui/handlers/process_handlers/pooling_process.py

```diff
@@ -114,13 +115,29 @@ def make_2D_arrays(plate, quant_process):
     layout = plate.layout
     raw_concs = np.zeros_like(layout, dtype=float)
     comp_concs = np.zeros_like(layout, dtype=float)
+    comp_is_blank = np.zeros_like(layout, dtype=bool)
+    plate_names = np.empty_like(layout, dtype='object')
     for comp, raw_conc, conc in quant_process.concentrations:
         well = comp.container
         row = well.row - 1
         column = well.column - 1
         raw_concs[row][column] = raw_conc
         comp_concs[row][column] = conc
-    return raw_concs, comp_concs
+
+        if isinstance(comp, LibraryPrep16SComposition):
```

Comment:
Both here and in the other if below, can you create a variable to store the 'sample_composition' element? Since you have to access that twice, recovering it once and accessing its properties would be faster, since you're doing less queries to the DB (you will do (n/2) + 1 calls instead of n)

Reply:
👍 For sure, totally missed this!